### PR TITLE
styling change to "boxed" view

### PR DESCRIPTION
**Pull request by dodosaurus**
image

Looks something like this, I tested it on different viewports, formatting looked OK. If you will like it, I can refactor it more and create some more readable global tailwind styles to have it more maintainable.

BTW thanks for this plugin, it is pure gold.

### CODE REVIEW

**Comment by filiphric** [comment]
wow, thanks for the contribution! I think this is a solid idea and I really liked the boxed view 👍 I have a couple of nitpicks before I merge:

- the font "inconsistency" is actually by design - I want to differentiate between headings and data, the same way it is done in Cypress timeline. all the "heading info" like test name, beforeEach hooks etc are written in sans, and all the commands are in monospace font. I’d like to keep it this way in the plugin as well so the UI feels unified with the timeline
- I made some suggestions in the code as well, especially for status box and `<hr>` styling

File: src/components/App.vue [hunk 1]
```diff
@@ -6,7 +6,7 @@
       <RequestHeaders :body="item.requestHeadersFormatted" />
       <Request :body="item.requestBodyFormatted" />
       <Response :body='item.responseBodyFormatted' :raw="item.responseBody"/>
-      <hr class="border-gray-300 border-1 my-2" />
+      <hr class="border-slate-400 border-dashed border-2 my-4" />
```
Comment (filiphric) [inline src/components/App.vue:9]: I’m not a fan of the dashed border, but I like the idea of changing it. I was thinking maybe something like this?
```html
<hr class="border-gray-300 border-1 my-4 w-3/4 m-auto" />
```

File: src/components/Title.vue [hunk 2]
```diff
@@ -1,10 +1,22 @@
...
+      <p class="font-mono font-bold">Method: </p>
+      <p class="px-2 ml-3 rounded-md inline-block text-white font-mono" :class="methodColor(method)">{{ method }}</p>
+    </div>
+
+    <div class="flex rounded-sm outline outline-slate-400 outline-1 p-2 mb-2">
+      <p class="font-mono font-bold">URL: </p>
+      <p class="ml-3 inline-block font-mono">{{ url }}</p>
+    </div>
+
+    <div class="flex rounded-sm outline outline-slate-400 outline-1 p-2 mb-2">
```
Comment (filiphric) [inline src/components/Title.vue:13]: I’d suggest showing this conditionally, so we see the stats box only when we have status:
```html
<div class="flex rounded-sm outline outline-slate-400 outline-1 p-2 mb-2" v-if="status">
```

**Review by filiphric — CHANGES_REQUESTED** [review]

**Comment by filiphric** [comment]
>If you will like it, I can refactor it more and create some more readable global tailwind styles to have it more maintainable.
oh YES! This is really a "quick and dirty" solution on my part. so some maintenance-friendly refactoring is definitely welcomed 👏

File: src/components/App.vue (hunk 1 above)
Comment (dodosaurus) [reply src/components/App.vue:9]: Yep, that thick dashed divider didn't look right. I tried to use your solution, but if we want to keep the boxed design, there are too much borders/lines imo, not very UX friendly. More viable design ideas:

1. We can logically box each request without outline, only with background
image
2. Do the same, but also add divider
image

What do you think?

Comment (dodosaurus) [reply src/components/App.vue:9]: btw, I also removed that "slate" color from everywhere, and we are only using now the gray color in different saturations.

Comment (filiphric) [reply src/components/App.vue:9]: I think the one with the background and without the `<hr>` line looks good 👍

Comment (filiphric) [reply src/components/App.vue]: seems to me the shadow is unnecessary, but otherwise this looks awesome 👍

**Comment by dodosaurus** [comment]
What was changed based on comments above:
- "font-mono" is no more used in titles, I used "font-semibold" not just default regular font to highlight title a little
- outlines were removed from everywhere, instead just background, gray gradients for colors are used to box the content into logical sections
- "Status" block is now conditional

image

File: src/components/App.vue [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
+  <div class="bg-transparent mb-10">
+    <div v-for="item in props.value">
+      <section class="bg-gray-100 rounded-sm m-4 p-4">
+        <Title :method="item.method" :url="item.url" :status="item.status" />
+        <Query :body="item.queryFormatted" />
+        <RequestHeaders :body="item.requestHeadersFormatted" />
+        <Request :body="item.requestBodyFormatted" />
+        <Response :body='item.responseBodyFormatted' :raw="item.responseBody" />
+      </section>
+      <!-- <hr class="border-gray-300 border-1 my-4 w-3/4 m-auto" /> -->
```
Comment (filiphric) [inline src/components/App.vue:11]: Feel free to remove this completely

File: src/components/Query.vue [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <template>
-  <div v-if="body" data-cy="query" class="mb-4">
-    <b>Query:</b>
+  <div class="bg-gray-200 rounded-sm p-2 mb-2" v-if="body" data-cy="query">
```
Comment (filiphric) [inline src/components/Query.vue:2]: we are reaching pixel-perfect design 😂 one more thing I noticed when using this boxed view, the `mb-2` class creates a bigger gap at the bottom of the box, so maybe going with something like `my-1` for all the boxes would work a little better
Screenshot_2022-10-11_at_11_26_08

**Review by filiphric — COMMENTED** [review]
Just to small nitpicks, but otherwise this is good to go

**Review by filiphric — APPROVED** [review]

**Pull request merged**